### PR TITLE
Feature/4402/check url

### PR DESCRIPTION
**Pull request by garyluu**
For dockstore/dockstore#4402

Add a simple field to the version info card for workflows.

If it's null, the field is not present

image

Suggestions welcome (always show field?, reword field?, reword tooltip?)

Test is going to fail for now for unrelated reasons

### CODE REVIEW

File: src/app/workflow/info-tab/info-tab.component.html [hunk 1]
```diff
@@ -283,6 +283,10 @@
         </a>
         <span *ngIf="!selectedVersion?.email"> n/a </span>
       </div>
+      <div *ngIf="publicAccessibleTestParameterFile !== null && publicAccessibleTestParameterFile !== undefined">
+        <strong matTooltip="Version has a test parameter file where all URLs are publicly available">Publicly Acessible Test Parameter File</strong>:
```
[the remark this answers is not in the document]
Comment (NatalieEO) [reply src/app/workflow/info-tab/info-tab.component.html]: I like the first option `Test parameter file with open data` and I think it's good to hide the field if it's null.

**Review by coverbeck — COMMENTED** [review]
Two thoughts based on your screenshot:

1. If we go with this approach, I feel a little "eye candy" would be nice (green checkmark, red x), although I guess it couldn't be done in isolation and may entail revisiting the whole layout... Does Kim already have a mock we haven't done for this page?
2. The checkUrl doesn't currently surface this information, and I brought this up on that PR, but I still feel it would be valuable to know which test parameter files have all public urls. If I'm a workflow user and find a workflow where this it says I have a all-public URLs parameter file, what do I do if there's more than one?

You could argue that's the job of the checker workflow and/or the description, and maybe that's the answer in combination with 1.

**Comment by garyluu** [comment]
> Two thoughts based on your screenshot:
> 
> 1. If we go with this approach, I feel a little "eye candy" would be nice (green checkmark, red x), although I guess it couldn't be done in isolation and may entail revisiting the whole layout... Does Kim already have a mock we haven't done for this page?
> 2. The checkUrl doesn't currently surface this information, and I brought this up on that PR, but I still feel it would be valuable to know which test parameter files have all public urls. If I'm a workflow user and find a workflow where this it says I have a all-public URLs parameter file, what do I do if there's more than one?
> 
> You could argue that's the job of the checker workflow and/or the description, and maybe that's the answer in combination with 1.

1. Unclear. But I personally don't consider that to be much of an enhancement. It would be an enhancement if it was in the versions table where you can at a glance compare versions really fast. Could you explain your reasoning?
2. There's an issue for this already https://github.com/dockstore/dockstore/issues/4384 . You can add that use case if you want. But for me (if I'm a workflow user even though I'm not), I would see that it's not publicly accessible and just give up. If there's more than one test parameter file (which is very unlikely), it should be pretty clear at a glance which test parameter file is publicly accessible. But let's not deal with hypotheticals for now, we have plenty of issues created by actual users with real use-cases that take higher priority.

**Review by denis-yuen — APPROVED** [review]

**Review by NatalieEO — APPROVED** [review]

**Comment by coverbeck** [comment]
>     1. Unclear. But I personally don't consider that to be much of an enhancement. It would be an enhancement if it was in the versions table where you can at a glance compare versions really fast. Could you explain your reasoning?

I just think it might look better, but don't feel too strongly about it.

**Review by coverbeck — APPROVED** [review]

**Pull request merged**